### DISCUSSION TITLE
"Geolocation source"-node

### DISCUSSION THREAD

**Discussion by sgofferj** [zachowj/node-red-contrib-home-assistant-websocket#432] [Feature Request]
Just like the entity node, it would be great to have a node which could create a geolocation source and geolocation objects from a list.

**Comment by zachowj** [comment]
Not sure I fully understand what you're looking for. Are you looking for the ability to create `device_tracker`s in HA from NR?

**Reply by sgofferj** [reply]
Basic description is here: https://www.home-assistant.io/integrations/geo_location/
There is e.g the Blitzortung integration which creates a geolocation source or included with HA is the geo_json integration:
https://www.home-assistant.io/integrations/geo_json_events/
https://github.com/home-assistant/core/tree/dev/homeassistant/components/geo_json_events

Basically, what I'm suggesting is a node which creates a geolocation source with a configurable name (like "blitzortung" or "geo_json_events"). The input would take a list/array object whose items must have a name, a lat and a lon attribute and maybe optionally a description attribute. Every time, the node would get the input, it would add the new items to the source and remove old ones which are not longer in the input object and fire the appropriate events.

The background for the feature suggestion is that there are plenty (useful) sources of data in the net which provide some kind of geo location information but not as geojson. With such a node, it would be possible to feed them into HA, display them on the HA map and use them in automations.
